src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -2601,8 +2601,9 @@ private void EnterScope(WriterState newState, ODataItem item)
                                 if (serializationInfo.NavigationSourceName != null)
                                 {
                                     ODataUriParser uriParser = new ODataUriParser(model, new Uri(serializationInfo.NavigationSourceName, UriKind.Relative), this.outputContext.Container);
-                                    odataUri = uriParser.ParseUri();
-                                    navigationSource = odataUri.Path.NavigationSource();
+                                    ODataUri odataUri = uriParser.ParseUri();
+                                    odataUriSlim = new ODataUriSlim(odataUri);
+                                    navigationSource = odataUriSlim.Path.NavigationSource();

```

Comment:
```suggestion
                                    ODataUri odataUri = new ODataUriSlim(uriParser.ParseUri());
                                    navigationSource = odataUri.Path.NavigationSource();
```
I'd also be curious whether `uriParser.ParseUri()` can return a null...

Reply:
`uriParse.ParseUri` does not return null. It creates a new instance of `ODataUri` directly and returns it.